File: C/0766.c
#include <stdio.h>

bool isToeplitzMatrix(int** matrix, int matrixSize, int* matrixColSize) {
    for (int i = 0; i < matrixSize; i++) {
        int r = i, c = 0;
        int val = matrix[r][c];
        r++, c++;

        while (r < matrixSize && c < *matrixColSize) {
            if (matrix[r][c] != val) {
                return 0;
            }
            r++, c++;
        }
    }

    for (int i = 0; i < *matrixColSize; i++) {
        int r = 0, c = i;
        printf("c = %d\n", c);
        int val = matrix[r][c];
        r++, c++;

        while (r < matrixSize && c < *matrixColSize) {
            if (matrix[r][c] != val) {
                return 0;
            }
            r++, c++;
        }
    }

    return 1;
}

File: Python/0766.py
class Solution:
    def isToeplitzMatrix(self, matrix: List[List[int]]) -> bool:
        rows = len(matrix)
        cols = len(matrix[0])

        # Check diagonals that start from top row
        for r in range(rows):
            c = 0
            val = matrix[r][c]
            
            r += 1
            c += 1
            while c < cols and r < rows:
                if matrix[r][c] != val:
                    return False
                r += 1
                c += 1
        
        # Check diagonals that start from left col
        for c in range(cols):
            r = 0
            val = matrix[r][c]
            
            r += 1
            c += 1
            while r < rows and c < cols:
                if matrix[r][c] != val:
                    return False
                r += 1
                c += 1

        return True
